Formula/coredns.rb

```diff
@@ -0,0 +1,57 @@
+class Coredns < Formula
+  desc "DNS server that chains plugins"
+  homepage "https://coredns.io/"
+  url "https://github.com/coredns/coredns/archive/v1.6.9.tar.gz"
+  sha256 "e100a946d5d60d936cb313bdb04fd96162d5a7229e08887ad497660acee4b36f"
+  head "https://github.com/coredns/coredns.git"
+
+  depends_on "go" => :build
+
+  def install
+    system "make"
+    bin.install "coredns"
+  end
+
+  plist_options :startup => true
+
+  def plist
+    <<~EOS
+      <?xml version="1.0" encoding="UTF-8"?>
+      <!DOCTYPE plist PUBLIC "-//Apple Computer//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+      <plist version="1.0">
+        <dict>
+          <key>Label</key>
+          <string>#{plist_name}</string>
+          <key>ProgramArguments</key>
+          <array>
+            <string>#{opt_bin}/coredns</string>
+            <string>-conf</string>
+            <string>#{etc}/coredns/Corefile</string>
+          </array>
+          <key>RunAtLoad</key>
+          <true/>
+          <key>KeepAlive</key>
+          <true/>
+          <key>StandardErrorPath</key>
+          <string>#{var}/log/coredns.log</string>
+          <key>StandardOutPath</key>
+          <string>#{var}/log/coredns.log</string>
+          <key>WorkingDirectory</key>
+          <string>#{HOMEBREW_PREFIX}</string>
+        </dict>
+      </plist>
+    EOS
+  end
+
+  test do
+    pid = Process.spawn("#{bin}/coredns -dns.port=6053")
+    ohai "Starting CoreDNS (PID #{pid})"
+    sleep(2)
+    output = shell_output("dig @127.0.0.1 -p 6053 example.com.")
+    assert_match(/example\.com\.\t\t0\tIN\tA\t127\.0\.0\.1\n/, output)
+  ensure
+    ohai "Stopping CoreDNS (PID #{pid})"
+    Process.kill("SIGINT", pid)
+    Process.wait(pid)
+  end
```

Comment:
This gets a random port, also the `ensure` isn't needed anymore.
```suggestion
    port = free_port
    pid = fork do
      exec bin/"coredns", "-dns.port=#{port}"
    end
    sleep(2)
    output = shell_output("dig @127.0.0.1 -p #{port} example.com.")
    assert_match(/example\.com\.\t\t0\tIN\tA\t127\.0\.0\.1\n/, output)
```